gcloud/storage/_implicit_environ.py

```diff
@@ -81,29 +29,11 @@ class _DefaultsContainer(object):
     :param connection: Persistent implied connection from environment.
     """
 
-    @_lazy_property_deco
-    @staticmethod
-    def project():
-        """Return the implicit default project."""
-        return _determine_default_project()
-
-    def __init__(self, project=None, bucket=None, connection=None,
-                 implicit=False):
-        if project is not None or not implicit:
-            self.project = project
+    def __init__(self, bucket=None, connection=None):
```

Comment:
Yes but `set_defaults` signal happens in `__init__.py`, which is back to my original concern.

I don't think it's cut and dry in either direction, I'm still torn.

Another concern has cropped up. What if they want to use a different default project for storage than the one used for pubsub?